stixcore/products/common.py

```diff
@@ -278,3 +280,63 @@ def rebin_proportional(y1, x1, x2):
     y2 += np.where(different_cell, contrib, 0.)
 
     return y2
+
+
+def unscale_triggers(scaled_triggers, *, integration, detector_masks, ssid, factor=30):
+    r"""
+    Unscale scaled trigger values.
+
+    Trigger values are scaled on board in compression mode 0,0,7 via the following relation
+
+    T_s = T / (factor * n_int * n_trig_groups)
+
+    where `factor` is a configured parameter, `n_int` is the duration in units of 0.1s and
+    `n_trig_groups` number of active trigger groups being summed, which depends on the data
+    product given by the SSID.
+
+    Parameters
+    ----------
+    scaled_triggers : int
+        Scaled trigger
+    integration : `astropy.units.Quantity`
+        Number of integrations (integration time / 0.1s)
+    detector_masks : `array-like`
+        Number of trigger groups summed
+    ssid : `int`
+        Science Structure ID
+    factor : int optional
+        Scaling factor
+    Returns
+    -------
+    tuple
+        Unscaled triggers, Scaling Variance
+    """
+    # TODO extract this to a separate function and test
+    detector_to_trigger_group_map = np.array(
+        [[1,  6,  5, 12, 14, 10,  8,  3, 31, 26, 22, 20, 18, 17, 24, 29],
+         [2,  7, 11, 13, 15, 16,  9,  4, 32, 27, 28, 21, 19, 23, 25, 30]]).T
+
+    trigger_groups = detector_masks[:, [np.array(detector_to_trigger_group_map) - 1]]
```

Comment:
is this not defined in any other place CONF ...

Reply:
Yea it probably should be but at the moment this is the only place it's used if we merged the full conf from the STIX-GSW and STIX-CONF it would all be in one place.